### DISCUSSION TITLE
Using statements in C#

### DISCUSSION THREAD

**Discussion by bigboybamo** [codewars/codewars.com#2440] [General]
A lot of times i want to solve katas but i have to manually type most of my using statements( Linq, generic) because im used to the .Net environment.

Is there a way to permanently have those statements loaded up or is this a feature the codewars team have to implement?

**Comment by hobovsky** [comment]
> Is there a way to permanently have those statements loaded up or is this a feature the codewars team have to implement?

It's neither. You simply need to type in all the needed code :)

It's not really feasible to put all potentially useful usings in initial solution stub of every kata, because author cannot possibly know what classes you would like to use in your solution.

If this feature would be configured per user, you would probably quickly notice that there are some usings you need to type in manually anyway, and many configured ones end up not being used that often as you thought they would be. I think that maintaining a list of "default usings" would quickly become tedious.

On a top of this, adding required usings is just one keyboard shortcut in an IDE. `Ctrl + .` is a really great command.

**Reply by bigboybamo** [reply]
alright then.

Does this shortcut work in the Codewars IDE?

**Reply by DeoEsor** [reply]
I don't think so.

**Comment by konrad-gora** [comment]
IMO it can be somehow solved.
`<ImplicitUsings>enable</ImplicitUsings>` should be added to csproj (in PropertyGroup section). It globally adds some basic usings. There is little risk that it can break actual solutions (if somebody creates a class that already exists in these usings), but this line can be added for C# 12.

I think also it would be good to add `<Nullable>enable</Nullable>` to make working code that uses nullable reference types.
